synapse/daemon.py

```diff
@@ -429,24 +431,47 @@ def _loadSessConf(self, info):
 
             self.onfini( core.fini )
 
+    def _onTelePushFiniMesg(self, sock, mesg):
+        name = mesg[1].get('name')
+        guid = mesg[1].get('guid')
+        current_guid = self.pushed_guids.get(name, None)
+
+        # user = sock.get('syn:user')
+        # if not self._isUserAllowed(user, 'tele:push:fini:'+name):
+        #     return
+        #     # XXX Not certain what sould be expected here since we're
+        #     # Firing this from the fini() handler of the Proxy object
+        #     # return sock.tx( tufo('job:done', err='NoSuchRule', jid=jid) )
+
```

Comment:
there's a check/set race here.  for now, i'd add a pushlock that you can hold for these little ops ( i have an idea brewing about a synapse global lock that we can use for small data structure access ( nearly guarenteed to be atomic by the GIL / scheduler ) but it's not priority yet. :D